tests/NATS.Client.Core.Tests/TlsClientTest.cs

```diff
@@ -22,6 +22,25 @@ public async Task Client_connect_using_certificate()
         Assert.True(rtt > TimeSpan.Zero);
     }
 
+    [Fact]
+    public async Task Client_connect_using_certificate_and_revocation_check()
+    {
+        await using var server = NatsServer.Start(
+            new NullOutputHelper(),
+            new NatsServerOptsBuilder()
+                .UseTransport(TransportType.Tls, tlsVerify: true)
+                .Build());
+
+        var clientOpts = server.ClientOpts(NatsOpts.Default with { Name = "tls-test-client" });
+        clientOpts = clientOpts with { TlsOpts = clientOpts.TlsOpts with { CheckCertificateRevocation = true } };
+        await using var nats = new NatsConnection(clientOpts);
+
+        // At the moment I don't know of a good way of checking if the revocation check is working
```

Comment:
We should probably try and come up with a real test.  If it's not supported in .NET 6 then we should probably conditionally compile the option.  I wonder how hard it'd be to use a self-signed CA that supports online revocations.

Reply:
> We should probably try and come up with a real test.

I don't really know how OCSP must-staple work. checked what mongodb driver is doing looked fairly involved to setup https://github.com/search?q=repo%3Amongodb%2Fmongo-csharp-driver%20ocsp&type=code but the c# code is just enabling this option.

> If it's not supported in .NET 6 then we should probably conditionally compile the option. I wonder how hard it'd be to use a self-signed CA that supports online revocations.

I was referring to ocsp stapling for .net6 AFAIU other types of online checks would still work. no need to exclude the option.

Reply:
> I was referring to ocsp stapling for .net6 AFAIU other types of online checks would still work. no need to exclude the option.

Oh good point